client/context.ts

```diff
@@ -107,21 +107,17 @@ export class Context {
     connectPacket: ConnectPacket,
   ): Promise<boolean> {
     this.mqttConn = new MqttConn({ conn });
-    if (this.mqttConn === undefined) {
-      return true;
-    }
-
     try {
       logger.debug("Send connect packet", connectPacket);
       await this.connect(connectPacket);
       logger.debug("Accepting packets");
       for await (const packet of this.mqttConn) {
-        handlePacket(this, packet);
```

Comment:
I assume we want to wait for the promise returned by `handlePacket()` to resolve, this signalling the end of the packet's processing, before moving to the next packet?